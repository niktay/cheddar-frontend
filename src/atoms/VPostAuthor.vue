<template>
  <span class="a-VPostAuthor">
    {{ byAuthor }}
  </span>
</template>

<script>
export default {
  computed: {
    byAuthor: function() {
      return `by ${this.author.firstName} ${this.author.lastName}`;
    }
  },

  props: {
    author: {
      type: Object,
      required: true,
      validator: function(value) {
        return (
          value.hasOwnProperty("firstName") && value.hasOwnProperty("lastName")
        );
      }
    }
  }
};
</script>

<style class="scss">
.a-VPostAuthor {
  color: #2b2b2b;
  font-family: "Roboto", sans-serif;
  font-size: 1rem;
  font-weight: 400;
  text-transform: uppercase;
}
</style>