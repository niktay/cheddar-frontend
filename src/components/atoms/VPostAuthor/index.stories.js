import { storiesOf } from "@storybook/vue";
import VPostAuthor from ".";

storiesOf("Atom - VPostAuthor", module).add("default", () => ({
  components: { VPostAuthor },

  data() {
    return {
      author: {
        firstName: "John",
        lastName: "Doe"
      }
    };
  },

  template: `<VPostAuthor :author="author"/>`
}));
