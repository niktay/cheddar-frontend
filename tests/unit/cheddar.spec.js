import { mount } from "@vue/test-utils";
import VPostAuthor from "../../src/atoms/VPostAuthor";

describe("VPostAuthor", () => {
  it("renders correctly setting the `author` prop", () => {
    const comp = mount(VPostAuthor, {
      propsData: {
        author: {
          firstName: "John",
          lastName: "Doe"
        }
      }
    });
    expect(comp.html()).toMatchSnapshot();
  });

  it("renders undefined author when no `author` prop", () => {
    const comp = mount(VPostAuthor);

    expect(comp.html()).toMatchSnapshot();
  });

  it("renders undefined for missing `author` prop properties", () => {
    const comp = mount(VPostAuthor, {
      propsData: {
        author: {}
      }
    });
    expect(comp.html()).toMatchSnapshot();
  });
});
