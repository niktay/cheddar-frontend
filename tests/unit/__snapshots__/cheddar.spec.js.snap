// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VPostAuthor renders correctly setting the \`author\` prop 1`] = `
<span class="a-VPostAuthor">
  by John Doe
</span>
`;

exports[`VPostAuthor renders undefined author when no \`author\` prop 1`] = `
<span class="a-VPostAuthor">
  by undefined undefined
</span>
`;

exports[`VPostAuthor renders undefined for missing \`author\` prop properties 1`] = `
<span class="a-VPostAuthor">
  by undefined undefined
</span>
`;
